### PR TITLE
update shell dependencies to prevent adding them on the extensions side since they are dashboard related 

### LINKED ISSUE #12319: Optimize dependencies on extensions related to the migration of vue3
**Is your feature request related to a problem? Please describe.**

With the progression of the work in the migration of vue3, which has reached a stability point as well the extensions compatibility side, there is room for optimisation on the dependencies we are adding to extensions since there's quite a few that could be included on the `shell` side instead of being added to the extension side

Ex: Stackstate https://github.com/StackVista/rancher-extension-stackstate/pull/22 or https://github.com/aalves08/rancher-extension-stackstate/blob/migrate-vue3/package.json#L15-L39

**Describe the solution you'd like**

Optmise dependencies added to every extension

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #12319 

### Occurred changes and/or fixed issues

- update shell dependencies to prevent adding them on the extensions side since they are dashboard related 
- bump shell version

I have some doubts in relation to:
`cache-loader`, since on the extensions migration we use `^4.1.0` and root dashboard uses `4.1.0`. Should we also pin it?
`node-polyfill-webpack-plugin`, since on the extensions migration we use `^3.0.0` and root dashboard uses `3.0.0`. Again, should we also pin it?

I've noticed quite a few resolutions that were already present on `shell` so I've only added the ones missing

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Comment by aalves08** [comment]
@jordojordo this will introduce changes on the migration script 🤔

File: shell/package.json [hunk 1]
```diff
@@ -153,7 +158,9 @@
     "qs": "6.11.1",
     "roarr": "7.0.4",
     "semver": "7.5.4",
-    "@vue/cli-service/html-webpack-plugin": "^5.0.0"
+    "@types/lodash": "4.17.5",
```
Comment (cnotv) [inline shell/package.json:161]: Other resolutions were not needed, they have been removed as unnecessary.

File: shell/package.json [hunk 2]
```diff
@@ -105,6 +109,7 @@
     "marked": "4.0.17",
     "nodemon": "2.0.22",
     "nyc": "15.1.0",
+    "node-polyfill-webpack-plugin": "^3.0.0",
```
Comment (cnotv) [inline shell/package.json:112]: You are right, I should have been pinning the versions as well. Would you mind to do that? 😬

File: shell/package.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
     "@rancher/icons": "2.0.29",
     "@types/is-url": "1.2.30",
     "@types/node": "16.4.3",
+    "@types/semver": "^7.5.8",
```
Comment (cnotv) [inline shell/package.json:44]: I don't think we need this, or?

Comment (aalves08) [reply shell/package.json:44]: I think we do @cnotv , as Stackstate extension is not using the `semver` library explicitly but it throws error on build related to that

Comment (cnotv) [reply shell/package.json:44]: Let's keep it then. Thanks for the update!

**Review by cnotv — APPROVED** [review]
LGTM

**Comment by jordojordo** [comment]
@aalves08 This shouldn't change anything on the migration script, we're only updating the [dependencies listed here](https://github.com/rancher/dashboard/blob/af3cc8f1f506b27144d0cdc696db15d74f92dd6f/creators/extension/migrate/tasks/packageUpdates.js#L68-L100) _if_ they already are present in the package.json.

**Review by jordojordo — APPROVED** [review]

**Pull request merged**